main/lsp/requests/completion.cc

```diff
@@ -251,37 +251,39 @@ vector<core::LocalVariable> allSimilarLocals(const core::GlobalState &gs, const
 
 string methodSnippet(const core::GlobalState &gs, core::SymbolRef method, core::TypePtr receiverType,
                      const core::TypeConstraint *constraint) {
-    auto shortName = method.data(gs)->name.data(gs)->shortName(gs);
-    vector<string> typeAndArgNames;
+    fmt::memory_buffer result;
+    fmt::format_to(result, "{}", method.data(gs)->name.data(gs)->shortName(gs));
+    auto nextTabstop = 1;
 
-    int i = 1;
+    vector<string> typeAndArgNames;
     if (method.data(gs)->isMethod()) {
         for (auto &argSym : method.data(gs)->arguments()) {
-            string s;
+            fmt::memory_buffer argBuf;
```

Comment:
creating `fmt::memory_buffer` is pretty costly. Could we use `result` instead?

Reply:
I was hoping to use `fmt::join` to handle putting the `, ` in the right places.

If i use `result`, i'll have to insert the logic to handle the commas myself.

Reply:
I tried this out and it's making the code less clear / more brittle and we don't even know whether this is a performance sensitive path yet (it only happens when generating the snippet for a completion item, which there are usually few of).

I'm going to go forward with the existing implementation for now and circle back to this when we come to tune performance of completion.